x-pack/plugins/index_management/public/hooks/use_details_page_mappings_model_management.ts

```diff
@@ -0,0 +1,139 @@
+
+    return { inferenceModels, trainedModelStats };
+  }, [api, ml]);
+
+  const getTrainedModelStats = useCallback(
+    (modelStats?: InferenceStatsResponse): DeploymentStatusType => {
+      const deploymentStatsByModelId =
+        modelStats?.trained_model_stats.reduce<DeploymentStatusType>((acc, modelStat) => {
+          if (modelStat.model_id) {
+            acc[modelStat.model_id] =
+              modelStat?.deployment_stats?.state === 'started' ? 'deployed' : 'not_deployed';
+          }
+          return acc;
+        }, {}) ?? {};
+
+      return deploymentStatsByModelId;
+    },
+    []
+  );
+
+  const getDefaultInferenceIds = useCallback((deploymentStatsByModelId: DeploymentStatusType) => {
+    return {
+      elser_model_2: {
+        trainedModelId: '.elser_model_2',
+        isDeployed: deploymentStatsByModelId['.elser_model_2'] === 'deployed',
+        defaultInferenceEndpoint: true,
+      },
+      e5: {
+        trainedModelId: '.multilingual-e5-small',
+        isDeployed: deploymentStatsByModelId['.multilingual-e5-small'] === 'deployed',
+        defaultInferenceEndpoint: true,
+      },
+    };
+  }, []);
+
+  const getCustomInferenceIdMap = useCallback(
+    (deploymentStatsByModelId: DeploymentStatusType, models?: InferenceModel) => {
+      return models?.data.reduce<InferenceToModelIdMap>((inferenceMap, model) => {
+        const inferenceId = model.model_id;
+        const trainedModelId =
+          'model_id' in model.service_settings ? model.service_settings.model_id : '';
+        inferenceMap[inferenceId] = {
+          trainedModelId,
+          isDeployed: deploymentStatsByModelId[trainedModelId] === 'deployed',
+          defaultInferenceEndpoint: false,
+        };
+        return inferenceMap;
+      }, {});
+    },
+    []
+  );
+
+  const getInferenceToModelIdMap = useCallback(async () => {
+    const { inferenceModels, trainedModelStats } = await fetchInferenceModelsAndTrainedModelStats();
+
+    const deploymentStatsByModelId = getTrainedModelStats(trainedModelStats);
+    const defaultInferenceIds = getDefaultInferenceIds(deploymentStatsByModelId);
+    const modelIdMap = getCustomInferenceIdMap(deploymentStatsByModelId, inferenceModels);
+    return { ...defaultInferenceIds, ...modelIdMap };
+  }, [
+    getCustomInferenceIdMap,
+    getDefaultInferenceIds,
+    getTrainedModelStats,
+    fetchInferenceModelsAndTrainedModelStats,
+  ]);
+
+  const getInferenceIdsInPendingList = useCallback(() => {
+    const denormalizedFields = deNormalize(state.fields);
+
+    const inferenceIdsInPendingList: Set<string> = new Set();
+    for (const field of Object.values(denormalizedFields)) {
+      if (field.type === 'semantic_text' && field.inference_id) {
+        inferenceIdsInPendingList.add(field.inference_id as string);
+      }
+    }
+
+    return inferenceIdsInPendingList;
+  }, [state.fields]);
+
+  const getPendingDeployments = useCallback(() => {
```

Comment:
This too should just be a const assignment, not a get function using a callback. Then you can also get rid of the `useState` up top. 

Note that this is derived state: it's just computed from existing data. We don't need to actively get data from an endpoint to figure this out, we just need to make sure that it's recomputed when the data it depends on is updated. React takes care of that for us. So this should be (assuming you updated `inferenceIdsInPendingList` above as well):

```
const pendingDeployments = useMemo(() => {
      return inferenceIdsInPendingList.map((inferenceId: string) => {
        const trainedModelId = inferenceToModelIdMap?.[inferenceId]?.trainedModelId ?? '';
        return (trainedModelId && !inferenceToModelIdMap?.[inferenceId]?.isDeployed ? trainedModelId : undefined;
      }).filter(trainedModelId => !!trainedModelId);
  }, [inferenceIdsInPendingList, inferenceToModelIdMap]);
});
```

Reply:
Got rid of useState and using useMemo instead of useCallback